### PR TITLE
Stock/update stockitem

### PR DESCRIPTION
**Pull request by ChristianBorn**

### CODE REVIEW

**Comment by Kreyno93** [comment]
Unfortunately i get a 405 on updating an item

**Review by Kreyno93 — DISMISSED** [review]

File: frontend/src/buttons/css/AbortButton.css [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+.abort-button:hover {
+    opacity: 0.7;
+    cursor: pointer;
+}
+
+.abort-button:active {
+    transition-duration: 0.1s;
+    opacity: 0.9;
+}
```
Comment (dennis-aclue) [inline frontend/src/buttons/css/AbortButton.css:21]: stupid, but new line!

File: frontend/src/icons/css/EditIcon.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.edit-record-icon {
+    font-size: 2em;
+    color: forestgreen;
+}
+
+.edit-record-icon :hover {
+    opacity: 0.8;
+}
```
Comment (dennis-aclue) [inline frontend/src/icons/css/EditIcon.css:8]: stupid, but need new line also.

File: frontend/src/stock/AddStockItemModal.tsx [hunk 3]
```diff
@@ -82,12 +82,13 @@ function AddStockItemModal(props: ModalProps) {
...
+                    </div>
                 </form>
             </section>
         </Modal>
     );
 }
 
-export default AddStockItemModal;
\ No newline at end of file
+export default AddItemModal;
```
Comment (dennis-aclue) [inline frontend/src/stock/AddStockItemModal.tsx:94]: same here with the new line

File: frontend/src/stock/EditItemModal.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                    <div className={"button-group"}>
+                        <input className={"submit-button"} type={"submit"} value={"Änderungen speichern"}/>
+                    </div>
+                </form>
+            </section>
+        </Modal>
+    );
+}
+
+export default EditItemModal;
```
Comment (dennis-aclue) [inline frontend/src/stock/EditItemModal.tsx:99]: new line!

File: frontend/src/stock/css/StockOverview.css [hunk 5]
```diff
@@ -43,10 +49,14 @@ tbody td:hover:before {
...
+    margin: 1.5em 0;
     color: #408900;
     font-weight: bold;
+    font-size: 1.2em;
+}
+
+.button-group {
+    display: flex;
+    justify-content: space-evenly;
 }
```
Comment (dennis-aclue) [inline frontend/src/stock/css/StockOverview.css:62]: new line please

File: frontend/src/stock/css/DeleteStockItemModal.css [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    margin: 0 auto;
+
+}
+
+span {
+    display: flex;
+    justify-content: center;
+    font-size: 1.2em;
+    font-weight: bold;
+}
```
Comment (dennis-aclue) [inline frontend/src/stock/css/DeleteStockItemModal.css:16]: new line please.

File: frontend/src/stock/css/AddItemModal.css [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    width: 50%;
+    height: 50%;
+    position: relative;
+    margin-right: auto;
+    margin-left: auto;
+}
+.ReactModal__Content--after-open section form {
+    display: flex;
+    flex-flow: column;
+}
```
Comment (dennis-aclue) [inline frontend/src/stock/css/AddItemModal.css:13]: new line please.

File: frontend/src/stock/EditItemModal.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    const [editedStockItem, setEditedStockItem] = useState<StockItemModel>(props.itemToEdit)
+    useEffect(() => setEditedStockItem(props.itemToEdit), [props.itemToEdit])
+
+
+    const saveNewStockitem = () => {
+        axios.put("/stock/overview", editedStockItem)
+            .catch((e) => console.error("PUT Error: " + e))
+            .then(props.reloadStockItems)
+            .then(props.closeModal)
+            .then(() => console.log(editedStockItem))
```
Comment (dennis-aclue) [inline frontend/src/stock/EditItemModal.tsx:29]: Do you need console log here?

File: frontend/src/icons/EditIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    return (
+        <span title={"Eintrag bearbeiten"} onClick={() => props.onClickAction(props.itemToEdit)}
+              className={"clickable-icon edit-record-icon"}>
+                    <FontAwesomeIcon icon={faPen}/>
+                            </span>
+
+    );
+}
+
+export default EditIcon;
```
Comment (dennis-aclue) [inline frontend/src/icons/EditIcon.tsx:23]: New line...

File: frontend/src/buttons/css/AbortButton.css (hunk 1 above)
Comment (ZeshanShahid79) [inline frontend/src/buttons/css/AbortButton.css:21]: pls add a new line here

File: frontend/src/icons/EditIcon.tsx (hunk 9 above)
Comment (ZeshanShahid79) [inline frontend/src/icons/EditIcon.tsx:23]: same here

File: frontend/src/stock/AddStockItemModal.tsx (hunk 3 above)
Comment (ZeshanShahid79) [inline frontend/src/stock/AddStockItemModal.tsx:94]: same here

**Review by dennis-aclue — DISMISSED** [review]

File: frontend/src/stock/EditItemModal.tsx [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+                    <div className={"button-group"}>
+                        <input className={"submit-button"} type={"submit"} value={"Änderungen speichern"}/>
+                    </div>
+                </form>
+            </section>
+        </Modal>
+    );
+}
+
+export default EditItemModal;
```
Comment (ZeshanShahid79) [inline frontend/src/stock/EditItemModal.tsx:98]: add a new line here as well :)

**Review by ZeshanShahid79 — COMMENTED** [review]
great job, add new lines as commented and it should be fine

**Review by dennis-aclue — APPROVED** [review]

**Pull request merged**